test/ipynb/latex/test_circuit_latex_drawer.py

```diff
@@ -15,99 +15,120 @@
-from .visualization import QiskitVisualizationTestCase
 
-pi = np.pi
-
-
-class TestLatexSourceGenerator(QiskitVisualizationTestCase):
-    """Qiskit latex source generator tests."""
-
-    def _get_resource_path(self, filename):
-        reference_dir = os.path.dirname(os.path.abspath(__file__))
-        return os.path.join(reference_dir, filename)
+RESULTDIR = os.path.dirname(os.path.abspath(__file__))
+
+
+@contextmanager
+def cwd(path):
+    """A context manager to run in a particular path"""
+    oldpwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(oldpwd)
+
+
+class TestLatexDrawer(QiskitTestCase):
+    """Testing Latex visualization."""
+
+    def setUp(self):
+        super().setUp()
+        self.circuit_drawer = TestLatexDrawer.save_data_wrap(_latex_circuit_drawer, str(self))
+
+    @staticmethod
+    def save_data_wrap(func, testname):
+        """A wrapper to save the data from a test"""
+
+        def wrapper(*args, **kwargs):
+            image_filename = kwargs["filename"]
+            with cwd(RESULTDIR):
+                results = func(*args, **kwargs)
+                TestLatexDrawer.save_data(image_filename, testname)
+            return results
+
+        return wrapper
+
+    @staticmethod
+    def save_data(image_filename, testname):
+        """Saves result data of a test"""
+        datafilename = "result_test.json"
+        if os.path.exists(datafilename):
+            with open(datafilename) as datafile:
+                data = json.load(datafile)
+        else:
+            data = {}
+        data[image_filename] = {"testname": testname}
+        with open(datafilename, "w") as datafile:
+            json.dump(data, datafile)
 
     def test_empty_circuit(self):
         """Test draw an empty circuit"""
-        filename = self._get_resource_path("test_latex_empty.tex")
         circuit = QuantumCircuit(1)
-        circuit_drawer(circuit, filename=filename, output="latex_source")
 
-        self.assertEqualToReference(filename)
+        self.circuit_drawer(circuit, filename="test_empty_circuit.png")
 
     def test_tiny_circuit(self):
         """Test draw tiny circuit."""
-        filename = self._get_resource_path("test_latex_tiny.tex")
         circuit = QuantumCircuit(1)
         circuit.h(0)
 
-        circuit_drawer(circuit, filename=filename, output="latex_source")
-
-        self.assertEqualToReference(filename)
+        self.circuit_drawer(circuit, filename="test_tiny_circuit.png")
 
     def test_normal_circuit(self):
         """Test draw normal size circuit."""
-        filename = self._get_resource_path("test_latex_normal.tex")
         circuit = QuantumCircuit(5)
         for qubit in range(5):
             circuit.h(qubit)
 
-        circuit_drawer(circuit, filename=filename, output="latex_source")
-
-        self.assertEqualToReference(filename)
+        self.circuit_drawer(circuit, filename="test_normal_circuit.png")
 
     def test_4597(self):
```

Comment:
This is not a very nice test name, we should have caught this earlier sorry! Would you ming renaming it as part of this PR?

Reply:
I also did a similar change in one of the mpl circuit tests. I hope that's fine.